### PR TITLE
Fix possible segfaults in ORCA. Batch 1

### PR DESCRIPTION
**Pull request by andr-sokolov**

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/64971

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1087423

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1087424

File: src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp [hunk 1]
```diff
@@ -214,6 +214,7 @@ CExpressionPreprocessor::PexprSimplifyQuantifiedSubqueries(CMemoryPool *mp,
 		1 == (*pexpr)[0]->DeriveMaxCard().Ull())
 	{
 		CExpression *pexprInner = (*pexpr)[0];
+		GPOS_ASSERT(NULL != pexprInner);
```
Comment (KnightMurloc) [inline src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp]: it looks like we are accessing `(*pexpr)[0]` before the assert.

**Review by RekGRpth — DISMISSED** [review]

File: src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp (hunk 1 above)
Comment (RekGRpth) [reply src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp]: This assert is need for below while cycle

Comment (RekGRpth) [reply src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp]: 109. here https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp#L220 pexprChild may be NULL via https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp#L221

Comment (KnightMurloc) [reply src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp]: yes, but we are accessing the same pointer in line 214.

Comment (RekGRpth) [reply src/backend/gporca/libgpopt/src/operators/CExpressionPreprocessor.cpp]: indeed, it should be reworked

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/65026

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1089823

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1089824

**Review by RekGRpth — APPROVED** [review]

**Pull request merged**